shell/src/main/java/alluxio/cli/docgen/ConfigurationDocGenerator.java

```diff
@@ -206,17 +223,82 @@ public static void writeYMLFile(Collection<? extends PropertyKey> defaultKeys, S
 
   /**
    * Generates the configuration docs.
+   * @param validate validate flag
    */
-  public static void generate() throws IOException {
+  public static void generate(boolean validate) throws IOException {
     Collection<? extends PropertyKey> defaultKeys = PropertyKey.defaultKeys();
     defaultKeys.removeIf(key -> key.isHidden());
     String homeDir = new InstancedConfiguration(ConfigurationUtils.defaults())
         .get(PropertyKey.HOME);
     // generate CSV files
     String filePath = PathUtils.concatPath(homeDir, CSV_FILE_DIR);
-    writeCSVFile(defaultKeys, filePath);
+    writeCSVFile(defaultKeys, filePath, validate);
+
+    if (validate) {
+      boolean hasDiff = false;
+      String[] fileNames = {"user-configuration.csv", "master-configuration.csv",
```

Comment:
this list is identical to the one in `writeCSVFile`. this should not be copied; if someone adds a new csv file, there's a good chance that it will be added onto one list but not the other. as much as possible, there should always be a "single source of truth", especially when there are constants involved